### PR TITLE
Feature: John The Ripper + Arm64 Support + Program Icon

### PR DESCRIPTION
**Pull request by PurpleSn0w**
- Added basic implementation of John The Ripper.

- Added dependencies for Arm64 support

- Added program icon

### CODE REVIEW

File: install_dependencies.sh [hunk 1]
```diff
@@ -23,19 +23,33 @@ sudo apt install -y mitmproxy \
...
+curl -fsSL https://deb.nodesource.com/setup_18.x | bash -
+exit
+E0F
 
 # Install openssl.
+echo "---------------------"
 echo "Installing openssl..."
+echo "---------------------"
 pushd /tmp
 wget http://nz2.archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.16_amd64.deb
```
Comment (Huskehhh) [inline install_dependencies.sh:54]: Remove the installation of volta now that nodesource is the install method of node

File: install_dependencies.sh [hunk 2]
```diff
@@ -23,19 +23,33 @@ sudo apt install -y mitmproxy \
         libjavascriptcoregtk-4.0-18 \
         libjavascriptcoregtk-4.0-dev \
         libwebkit2gtk-4.1 \
-        libwebkit2gtk-4.1-dev
+        libwebkit2gtk-4.1-dev \
+		cargo \
+		nodejs
```
Comment (Huskehhh) [inline install_dependencies.sh:28]: This needs to be done AFTER node source is installed

File: install_dependencies.sh [hunk 3]
```diff
@@ -50,18 +64,24 @@ volta setup
 export VOLTA_HOME="$HOME/.volta"
 export PATH="$VOLTA_HOME/bin:$PATH"
 
+echo "---------------------------"
 echo "Installing node and yarn..."
+echo "---------------------------"
 # Install node and yarn.
 volta install node
 volta install yarn
```
Comment (Huskehhh) [inline install_dependencies.sh:72]: Install yarn via NPM instead now that node source is being used

File: src-tauri/Cargo.toml [hunk 4]
```diff
@@ -17,7 +17,7 @@ tauri-build = { version = "1", features = [] }
 [dependencies]
 serde_json = "1.0"
 serde = { version = "1.0", features = ["derive"] }
-tauri = { version = "1", features = ["shell-execute", "shell-open", "shell-sidecar"] }
+tauri = { version = "1", features = ["fs-all", "path-all", "shell-all"] }
```
Comment (Huskehhh) [inline src-tauri/Cargo.toml:20]: Nice!

File: src/components/JohnTheRipper/JohnTheRipper.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                setOutput(result);
+            } catch (e: any) {
+                console.log(e);
+                setOutput(e);
+            }
+
+            try {
+                const args = [`--format=${values.fileType}`, "/tmp/hash.txt"];
+                const result = await CommandHelper.runCommand("john", args);
+                console.log(result);
```
Comment (Huskehhh) [inline src/components/JohnTheRipper/JohnTheRipper.tsx:48]: Remove this (and others similar)

**Review by Huskehhh — CHANGES_REQUESTED** [review]
Mostly looks great! Just needs a few minor changes for the ARM64 support

File: src/components/JohnTheRipper/JohnTheRipper.tsx (hunk 5 above)
Comment (PurpleSn0w) [reply src/components/JohnTheRipper/JohnTheRipper.tsx:48]: This is required. Unless there is a way to shorten the code.
In the if true statement
The first command find the hash and stores it. The second reads the hash and cracks the password.
In the if else statement
Reads the user inputed hash and cracks the password.

File: install_dependencies.sh (hunk 3 above)
Comment (PurpleSn0w) [reply install_dependencies.sh]: Made some adjustments and pushed an updated install_dependencies.sh

File: src/components/JohnTheRipper/JohnTheRipper.tsx (hunk 5 above)
Comment (Huskehhh) [reply src/components/JohnTheRipper/JohnTheRipper.tsx:48]: Sorry, meant the `console.log`

Comment (PurpleSn0w) [reply src/components/JohnTheRipper/JohnTheRipper.tsx]: Done

**Review by Huskehhh — APPROVED** [review]

**Pull request merged**